src/issuance.rs

```diff
@@ -83,27 +123,58 @@ impl IssueAction {
         self.finalize
     }
 
-    /// Return the `AssetBase` if the provided `ik` is used to derive the `asset_id` for **all** internal notes.
-    fn are_note_asset_ids_derived_correctly(
+    /// Computes the new asset supply for an `IssueAction`.
+    ///
+    /// This function calculates the total value of the asset by summing the values of all its notes
+    /// and ensures that all note types are equal. It returns the asset and its supply as a tuple
+    /// (`AssetBase`, `AssetSupply`) or an error if the asset was not properly derived or an
+    /// overflow occurred during the supply amount calculation.
+    ///
+    /// # Arguments
+    ///
+    /// * `ik` - A reference to the `IssuanceValidatingKey` used for deriving the asset.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing a tuple with an `AssetBase` and an `AssetSupply`, or an `Error`.
+    ///
+    /// # Errors
+    ///
+    /// This function may return an error in any of the following cases:
+    ///
+    /// * `IssueActionIncorrectAssetBase`: If the asset type of any note in the `IssueAction` is
+    ///   not equal to the asset type of the first note.
+    ///
+    /// * `ValueSumOverflow`: If the total amount value of all notes in the `IssueAction` overflows.
+    ///
+    /// * `IssueBundleIkMismatchAssetBase`: If the provided `ik` is not used to derive the
+    ///   `asset_id` for **all** internal notes.
+    fn compute_new_supply(
```

Comment:
let's rename it again into: `verify_supply()`, since it is critical to consensus verification with the extra benefit of supply compute.

Reply:
Done.